samza-core/src/main/java/org/apache/samza/zk/ZkMetadataStore.java

```diff
@@ -64,25 +69,47 @@ public void init() {
    */
   @Override
   public byte[] get(String key) {
```

Comment:
Please add unit tests for large get/put/delete and helper methods in this class.